compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/text/TextFieldDelegate.skiko.kt

```diff
@@ -0,0 +1,158 @@
+ * @param editProcessor The EditProcessor object that manages the editing operations of the TextField.
+ * @param offsetMapping The OffsetMapping object that maps the transformed offset to the original offset.
+ * @param showContextMenu The function that displays the context menu at the given rectangular area.
+ * @param onValueChange The function that is called when the TextField value changes.
+ */
+internal fun TextFieldDelegate.Companion.cupertinoSetCursorOffsetFocused(
+    position: Offset,
+    textLayoutResult: TextLayoutResultProxy,
+    editProcessor: EditProcessor,
+    offsetMapping: OffsetMapping,
+    showContextMenu: (Rect) -> Unit,
+    onValueChange: (TextFieldValue) -> Unit
+) {
+    val offset =
+        offsetMapping.transformedToOriginal(textLayoutResult.getOffsetForPosition(position))
+    val currentValue = editProcessor.toTextFieldValue()
+    val currentText = currentValue.text
+
+    val cursorDesiredOffset = determineCursorDesiredOffset(
+        offset,
+        currentValue,
+        textLayoutResult,
+        currentText
+    )
+
+    if (cursorDesiredOffset == offset) {
+        showContextMenu(textLayoutResult.value.getCursorRect(offset))
+    }
+    onValueChange(
+        editProcessor.toTextFieldValue().copy(selection = TextRange(cursorDesiredOffset))
+    )
+}
+
+/**
+ * Determines the desired cursor position based on the given parameters.
+ *
+ * The rules for determining position of the caret are as follows:
+ * - When you make a single tap on a word, the caret moves to the end of this word.
+ * - If there’s a punctuation mark after the word, the caret is between the word and the punctuation mark.
+ * - If you tap on a whitespace, the caret is placed before the word. Same for many whitespaces in a row. (and punctuation marks)
+ * - If there’s a punctuation mark before the word, the caret is between the punctuation mark and the word.
+ * - When you make a single tap on the first half of the word, the caret is placed before this word.
+ * - If you tap on the left edge of the TextField, the caret is placed before the first word on this line. The same is for the right edge.
+ * - If you tap at the caret, that is placed in the middle of the word, it will jump to the end of the word.
+ * @param offset The current offset position.
+ * @param currentValue The current TextFieldValue.
+ * @param textLayoutResult The TextLayoutResultProxy representing the layout of the text.
+ * @param currentText The current text in the TextField.
+ * @return The desired cursor position after evaluating the given parameters.
+ */
+@OptIn(ExperimentalTextApi::class)
+internal fun determineCursorDesiredOffset(
+    offset: Int,
+    currentValue: TextFieldValue,
+    textLayoutResult: TextLayoutResultProxy,
+    currentText: String
+): Int {
+    val caretOffsetPosition: Int
+    if (isCaretTapped(offset, currentValue.selection.start)) {
```

Comment:
```suggestion
    val previousCaretOffset = currentValue.selection.start
    if (offset == previousCaretOffset) {
```

The function that only compares 2 values looks redundant. If we want to add meaning to the values, I think better to do that this way.

Reply:
done